src/Tribe/Aggregator/Service.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+// Don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+// bail if already defined
+if ( class_exists( 'Tribe__Events__Aggregator__Service' ) ) {
+	return;
+}
+
+class Tribe__Events__Aggregator__Service {
+	/**
+	 * @var Tribe__Events__Aggregator__Service Event Aggregator Service class
+	 */
+	protected static $instance;
+
+	/**
+	 * @var Tribe__Events__Aggregator Event Aggregator object
+	 */
+	protected $aggregator;
+
+	/**
+	 * @var string Event Aggregator API version
+	 */
+	protected $api_version = 'v1';
+
+	/**
+	 * @var string Event Aggregator URL
+	 */
+	protected $api_base_url = 'http://api.tri.be/';
+
+	/**
+	 * @var string Event Aggregator API root
+	 */
+	protected $api_root = 'wp-json/event-aggregator/';
+
+	/**
+	 * Static Singleton Factory Method
+	 *
+	 * @return Tribe__Events__Aggregator__Service
+	 */
+	public static function instance( Tribe__Events__Aggregator $aggregator ) {
+		if ( ! isset( self::$instance ) ) {
+			$className      = __CLASS__;
+			self::$instance = new $className( $aggregator );
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor
+	 */
+	public function __construct( $aggregator ) {
+		$this->aggregator = $aggregator;
+
+		if ( defined( 'EVENT_AGGREGATOR_API_BASE_URL' ) ) {
+			$this->api_base_url = EVENT_AGGREGATOR_API_BASE_URL;
+		}
+
+		$this->fetch_origins();
+	}
+
+	/**
+	 * Fetch origins from service
+	 *
+	 * @return array
+	 */
+	public function fetch_origins() {
+		$origins = array(
+			(object) array(
+				'id' => 'csv',
+				'name' => __( 'CSV File', 'the-events-calendar' ),
+			),
+		);
+
+		// if the user doesn't have a license key, don't bother hitting the service
+		if ( ! $key = $this->aggregator->get_license_key() ) {
+			return $origins;
+		}
+
+		$response = $this->get( $this->build_url( 'origin' ) );
+
+		if ( $response && 'success' === $response->status ) {
+			$origins = array_merge( $origins, $response->data->origin );
+		}
+
+		return $origins;
+	}
+
+	/**
+	 * Performs a GET request against the Event Aggregator service
+	 *
+	 * @param string $url Endpoint URL
+	 * @param array $data Array of parameters to send to the endpoint
+	 *
+	 * @return stdClass
+	 */
+	public function get( $url, $data = array() ) {
```

Comment:
Could `$url` be changed to `$endpoint` to avoid us having to call multiple methods all the time. Or maybe have a register method to setup all the possible endpoints ?